execution_workflows/DaPars2/Dapars2_snakemake/workflow/Snakefile

```diff
@@ -0,0 +1,287 @@
+    log:
+        os.path.join(config["local_log"], "getBed12_2.log")
+    shell:
+        "./scripts/genePredToBed {input.genePred} {output.out}"
+        
+rule extractGeneIdAndName:
+    """This step extracts gene ID and gene symbol from gtf.
+    """
+    input:
+        gtf=os.path.join(config["out_dir"], "annotation_Chr.gtf")
+    output:
+        out=os.path.join(config["out_dir"], "transcriptIdSymbol.txt")
+    log:
+        os.path.join(config["local_log"], "extractGeneIdAndName")
+    shell:
+        "python ./scripts/extractIdSymbol.py <{input.gtf} >{output.out}"
+
+#-------------------------------------------------------------------------------
+# Method-specific rules
+rule getRegionAnnotation:
+    """This step generate 3UTR region annotation.
+    """
+    input:
+        bed=os.path.join(config["out_dir"], "annotation_Chr.bed"),
+        symbol=os.path.join(config["out_dir"], "transcriptIdSymbol.txt")
+    output:
+        out=os.path.join(config["out_dir"], "extracted_3UTR.bed")
+    log:
+        os.path.join(["local_log"], "getRegionAnnotation")
+    conda:
+        "./envs/dapars2.yml"
+    shell:
+        "python2 ./scripts/DaPars_Extract_Anno.py -b {input.bed} -s {input.symbol} -o {output.out}"
+
+rule makeDapars2Config:
+    """This step makes configure file for DaPars2.
+    """
+    input:
+        bed=os.path.join(config["out_dir"], "annotation_Chr.bed"),
+        seqDepth=os.path.join(config["out_dir"], "pathReadcounts.tsv"),
+        bedgraph=os.path.join(config["out_dir"], "{sample}_Chr.bedgraph")
+    output:
+        out=os.join.path(config["out_dir"], "Dapars2_configure_file")
+    params:
+        ct=config["coverageThreshold"]
+    threads:
+        4
+    log:
+        os.path.join(config["local_log"], "makeDapars2Config")
+    run:
+        sample_list=""
+        for i in input.bedgraph:
+            name=str(i)[:-9]
+            sample_list=sample_list+name+","
+        sample_list=sample_list.strip(",")
+        with open({output}[0], "w") as f:
+            print("# The following file is the result of generate_region_annotation", file=f)
+            print("Annotated_3UTR="+str(input.bed), file=f)
+            print("A comma-separated list of bedgraph files of all samples", file=f)
+            print("Aligned_wig_files="+sample_list, file=f)
+            print("Output_directory=DaPars2_results", file=f)
+            print("Output_result_file=DaPars2_results", file=f)
+            print("Coverage_threshold="+{params.ct}, file=f)
+            print("Num_Threads = "+{threads}, file=f)
+            print("sequencing_depth_file="+{input.seqDepth})
+
+rule mainDapars2:
+    """This step runs DaPars2 on multiple samples and multiple chromosomes.
+    """
+    input:
+        out=os.join.path(config["out_dir"], "Dapars2_configure_file")
+    output:
+        # This step will generate "DaPars2_results+chr[1-22XY]" and each dir contains "DaPars2_results_temp.chr[1-22XY].txt"
```

Comment:
@mkatsanto @dominikburri can you help here?

Reply:
You could enumerate all output files since you know all chromosome names, resp. you can get them. From the top of my head, maybe list comprehension helps. Here an example for 3 chromosomes: `["DaPars2_results_temp." + chrN + ".txt" for chrN in ['chr1', 'chr2','chr3']]` (I didn't test it, so no guarantee that it is correct).
Then you can specify the output path in `params`, should be `os.path.join(config["out_dir"],  "DaPars2_result_chr[1-22XY]"`.
This parameter you can use in your scripts. This means that you pass explicitly only the directory to your call, but implicitly Snakemake expects all the listed output files. However, this is just a suggestion, maybe @mkatsanto has a better approach.